content/router.xql

```diff
@@ -307,8 +307,55 @@ declare function router:body ($request as map(*)) {
     else (
         try {
             switch ($request?format)
-            (: Q: do we need to handle form-data? :)
-            case "form-data" return () 
+            case "form-data" return (
+                if (request:is-multipart-content())
+                then (
+                    let $properties := $request?config?requestBody?content?("multipart/form-data")?schema?properties
+                    return map:merge(
+                        for $name in map:keys($properties)
+                        let $property := $properties?($name)
+                        let $is-array := $property?type = "array"
+                        let $format :=
+                            if ($is-array)
+                            then $property?items?format
+                            else $property?format
+                        return
+                            (: TODO: throw if not $is-array but more than one item received :)
```

Comment:
what about these todos? Solved or not and being crucial or not?

Reply:
Nice to have - I would say. But there is still plenty of improvements.
Also missing is the validation of each requested properties format (integer, ...) when populating the maps.
All part of the greater validation goal.

Reply:
What is also not implemented yet is the ability to limit uploads to certain file-/mimetypes which is part of the OpenAPI spec.